runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/interceptors/OperationTelemetryInterceptor.kt

```diff
@@ -94,6 +96,13 @@ internal class OperationTelemetryInterceptor(
         if (attempts > 1) {
             metrics.rpcRetryCount.add(1L, perRpcAttributes, metrics.provider.contextManager.current())
         }
+
+        val attemptDuration = attemptStart?.elapsedNow() ?: return
```

Comment:
> Change the implementation of smithy.client.attempt_duration to measure attempt duration instead of merely transmission duration

This now includes signing and other things right? The current description of the metric was accurate for what it measured: `The time it takes to connect to the service, send the request, and receive the HTTP status code and headers from the response for an operation`. If we are updating the metric definition lets update the description to match.

Can you first explain _why_ this is needed/better? Just want to understand the motivation for the change.

Reply:
Yes, this now includes identity resolution, endpoint resolution, and signing. The inclusion of EP resolution and signing will likely have minimal impact to the metric but identity resolution is a potentially-expensive operation since it may involve local IO or calls to remote services. Given that the intent is to measure the SDK-added overhead and that identity resolution, EP resolution, and signing are _per-attempt_, we'd need to know the duration of the _entire attempt_.

I'll update the metric documentation.